### ISSUE TITLE
v4-without-jquery EventHandler?

### ISSUE THREAD

**Issue by emilmuller** [twbs/bootstrap#28071]
v4-**without**-jquery has a Modal class.

How do I go about listening to [events](https://getbootstrap.com/docs/4.2/components/modal/#usage)? E.g. `show`, `hide`, etc.

From looking at the source code, it seems I need to access EventHandler. It is not exported. Should it be exported? I went ahead and added it to index.js, and recompiled:

```
import Alert from './alert'
import Button from './button'
import Carousel from './carousel'
import Collapse from './collapse'
import Dropdown from './dropdown'
import Modal from './modal'
import Popover from './popover'
import ScrollSpy from './scrollspy'
import Tab from './tab'
import Toast from './toast'
import Tooltip from './tooltip'
import Util from './util'
import EventHandler from './dom/eventHandler'

/**
 * --------------------------------------------------------------------------
 * Bootstrap (v4.2.1): index.js
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * --------------------------------------------------------------------------
 */

export {
  Util,
  Alert,
  Button,
  Carousel,
  Collapse,
  Dropdown,
  Modal,
  Popover,
  ScrollSpy,
  Tab,
  Toast,
  Tooltip,
  EventHandler
}
```

Not sure if this is the correct approach?

**Comment by Johann-S** [comment]
our EventHandler trigger native events and jQuery events so you can do the following:
```js
const modalNode = document.getElementById('myModal')
const modal = new Modal(modalNode)

modalNode.addEventListener('shown.bs.bootstrap', () => {
	console.log('modal shown')
})

**Comment by emilmuller** [comment]
@Johann-S thanks! I think so, but I'm having other issues as well.

When triggering show() on the Modal class, it throws an error:

```
script.js:11405 Uncaught (in promise) TypeError: $.Event is not a function
    at Object.trigger (script.js:11405)
    at Modal.show (script.js:14291)
```
Looking at the line in question leads me to line 262 in eventHandler.js:

```
if (inNamespace && typeof $ !== 'undefined') {
  // Evaluates to true eventhough jQuery does not exist (!)
```

Looking at Util.jQuery:

```
  get jQuery() {
    return window.$ || window.jQuery
  }
```

In my system, window.$ is actually defined without jQuery being loaded. https://stackoverflow.com/questions/22244823/what-is-the-dollar-sign-in-javascript-if-not-jquery

Maybe just check window.jQuery? :)

**Comment by emilmuller** [comment]
I'm afraid :fearful:
I'll see what I can manage :sweat_smile:

**Comment by Johann-S** [comment]
if you have question on how to do that, do not hesitate to hit me on Bootstrap's Slack 😉

**Comment by emilmuller** [comment]
I think I did it. Hope I didn't break bootstrap in the process :smile: #28074

**Comment by Johann-S** [comment]
Closed thanks to #28074
